integration/container/kill_test.go

```diff
@@ -0,0 +1,182 @@
+package container
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/api/types/strslice"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/require"
+)
+
+func TestKillContainerInvalidSignal(t *testing.T) {
+	defer setupTest(t)()
```

Comment:
I don't think this works with `t.Parallel()`
For what it's worth, the auto-cleanup is probably not needed as it's simple to cleanup manually.

Reply:
Good catch. Altough we don't actually use `-parallel` right now so these `t.Parallel()` don't do anything I believe.

Reply:
I think we should make the extra effort to have these run parallel correctly, as we move stuff over it will be a huge help to test run times.

Reply:
While trying to migrate some integration-cli tests to integration api tests, I noticed some of the tests rely on fixed "name" of containers, like `foo`, `server`, etc.. I think that may not work well with `parallel`. May need to convert them to random generated names to avoid collision.

Reply:
Ya, I'm not against trying to get things running in parallel, but it is a lot more work than just adding `t.Parallel()`. 

It's a lot easier to get the cleanup wrong if we don't have an easy way to label everything with the test name.

Reply:
> May need to convert them to random generated names to avoid collision.

Where possible we should just capture the container ID that's returned

Reply:
Yeah, I think we'd need to have the client automatically namespace objects or something then the defer could cleanup just the namespaces resources.